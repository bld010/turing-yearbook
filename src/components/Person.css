.Person {
  height: 350px;
  width: 220px;
  border: 1px solid black;
  background: blanchedalmond;
  margin: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  box-shadow: 2px 2px grey;
}

.Person img {
  max-width: 98%;
  max-height: 100px;
  border: 1px solid black;
}

.Person h4, 
.Person p {
  margin: 20px 10px 20px 10px;
  height: 30px;
}

.Person button {
  width: 15px;
  height: 15px;
  border-radius: 50%;
  padding: 0;
  text-align: center;
}

.Person div {
  width: 90%;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  margin: 5px 5px 10px 0px;
}
