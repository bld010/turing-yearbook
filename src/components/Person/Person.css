
.Person {
  height: 300px;
  width: 200px;
  border: 1px solid black;
  background: white;
  margin: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
  border-radius: 2px;
  box-shadow: 2px 2px 5px 2px #e1e1e1;
}

.Person img {
  max-width: 90%;
  max-height: 100px;
  border: 1px solid black;
}

.superlative img,
.quote img {
  border: 0px;
  height: 20px;
}

.quote,
.superlative {
  padding-left: 15px;
}

.quote>p {
  min-height: 30px;
}

.Person h4, 
.Person p {
  margin: 0px 10px 20px 10px;
  height: 20px;
  font-size: 14px;
  min-width: 70px;
}

.Person button {
  width: 18px;
  height: 18px;
  padding: 0;
  text-align: center;
  font-size: 14px;
  border: none;
}

.Person button:hover {
  cursor: pointer;
}

.Person div {
  width: 90%;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  margin: 5px 5px 10px 0px;
}

.Person h3 {
  margin: 10px 0 10px 0;
}

.staff>h3 {
  background-color: #8bf0ba;
  padding: 0 8px 0 8px;
  background-image: linear-gradient(to right, #8bf0ba, #b6f3d3);
}

.students>h3 {
  background-color:  #f2b1d8;
  padding: 0 8px 0 8px;
  background-image: linear-gradient(to right,  #f2b1d8,  #f5d1e8);
}