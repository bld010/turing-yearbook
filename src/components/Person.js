import React from 'react';
import './Person.css';

const Person = ({ person }) => {
  return (
    <div className="Person">
      <img src={person.photo} alt="Person"></img>
      <h3>{person.name}</h3>
      <p>"{person.quote}"</p>
      <h4>{person.superlative}</h4>
    </div>
  )
}
export default Person;
