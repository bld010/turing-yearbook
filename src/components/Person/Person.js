import React, { Component } from 'react';
import './Person.css';

class Person extends Component {
  constructor({ person, cohortType, deletePerson}) {
    super({ person, cohortType, deletePerson})
    this.state = {
      name: person.name,
      id: person.id,
      quote: person.quote,
      superlative: person.superlative,
      photo: person.photo,
      cohortType: cohortType, 
      deletePerson: deletePerson
    }
  }

handleEdits(e, partToUpdate) {
  this.setState({[e.target.id]: e.target.innerText})
}

handleCompletedChange = () => {
  this.props.updatePeople(this.state)
}
  
render() {
  return (
    <div className={this.state.cohortType + " Person"}>
      <div>
        <button onClick={() => this.state.deletePerson(this.state.id, this.state.cohortType)}>x</button>
      </div>
      <img src= {this.state.photo} alt="Person"></img>
      <h3 
        id="name" 
        contentEditable="true"
        suppressContentEditableWarning={true}
        onKeyUp={(e) => this.handleEdits(e, 'name')}
        onBlur={this.handleCompletedChange}>
        {this.state.name}
      </h3>
      <div className="quote">
        <img src={ require('./conversation.svg') } alt="quote bubble"></img>
        <p 
          id="quote" 
          contentEditable="true"
          suppressContentEditableWarning={true}
          onKeyUp={e => this.handleEdits(e)}
          onBlur={this.handleCompletedChange}>
          "{this.state.quote}"
        </p>
      </div>
      <div className="superlative">
        <img src={ require('./medal.svg') } alt="medal icon"></img>
        <h4 
          id="superlative" 
          contentEditable="true"
          suppressContentEditableWarning={true}
          onKeyUp={this.handleEdits}
          onBlur={this.handleCompletedChange}>
          {this.state.superlative}
        </h4>
      </div>
    </div>
  )
}
}
export default Person;
