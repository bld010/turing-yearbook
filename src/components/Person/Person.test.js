import React from 'react';
import { shallow } from 'enzyme';
import Person from './Person';

let wrapper = null;
let instance = null;
let person = {id: 8, name: 'Robbie', quote: 'Cool.', superlative: 'Most Likely to Be Accused of Murder', photo: './robbie.jpg'};

describe('Person', () => {
  beforeEach(() => {
    wrapper = shallow(<Person key={person.name} cohortType="staff" updatePople={jest.fn()} deletePerson={jest.fn()} person={person} />)
    instance = wrapper.instance();
  })
  it('should match the snapshot with all correct data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('Editing Contents', () => {
  
    it('should update state when name is edited', () => {
      const mockEvent = { target: {
        id: 'name', 
        innerText: 'Robert J.'
      }}
      wrapper.find('#name').simulate('keyUp', mockEvent);
      expect(instance.state.name).toEqual('Robert J.')
    });

    it('should update state when quote is edited', () => {
      const mockEvent = { target: {
        id: 'quote', 
        innerText: 'Yolo?'
      }}
      wrapper.find('#quote').simulate('keyUp', mockEvent);
      expect(instance.state.quote).toEqual('Yolo?')
    });

    it('should update state when superlative is edited', () => {
      const mockEvent = { target: {
        id: 'superlative', 
        innerText: 'Most Likely To Wear Birkenstocks'
      }}

      wrapper.find('#superlative').simulate('keyUp', mockEvent);
      expect(instance.state.superlative).toEqual('Most Likely To Wear Birkenstocks')
    })

    it('should fire handleCompletedChange when user clicks away after editing', () => {
      const mockHandleCompletedChange = jest.fn();
      instance.setState({handleCompletedChange: mockHandleCompletedChange});
      wrapper.find('#name').simulate('blur')
      expect(mockHandleCompletedChange).toHaveBeenCalled();
    })
  })

  it('should fire deletePerson when x is clicked',() => {
    let mockDeletePerson = jest.fn();
    instance.state.deletePerson = mockDeletePerson;
    wrapper.find('button').simulate('click');
    expect(mockDeletePerson).toHaveBeenCalled();
  })
})


