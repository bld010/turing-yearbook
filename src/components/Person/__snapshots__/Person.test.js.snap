// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should match the snapshot with all correct data passed in 1`] = `
<div
  className="staff Person"
>
  <div>
    <button
      onClick={[Function]}
    >
      x
    </button>
  </div>
  <img
    alt="Person"
    src="./robbie.jpg"
  />
  <h3
    contentEditable="true"
    id="name"
    onBlur={[Function]}
    onKeyUp={[Function]}
    suppressContentEditableWarning={true}
  >
    Robbie
  </h3>
  <div
    className="quote"
  >
    <img
      alt="quote bubble"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "conversation.svg",
        }
      }
    />
    <p
      contentEditable="true"
      id="quote"
      onBlur={[Function]}
      onKeyUp={[Function]}
      suppressContentEditableWarning={true}
    >
      "
      Cool.
      "
    </p>
  </div>
  <div
    className="superlative"
  >
    <img
      alt="medal icon"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "medal.svg",
        }
      }
    />
    <h4
      contentEditable="true"
      id="superlative"
      onBlur={[Function]}
      onKeyUp={[Function]}
      suppressContentEditableWarning={true}
    >
      Most Likely to Be Accused of Murder
    </h4>
  </div>
</div>
`;
